### DISCUSSION TITLE
Memory usage problem

### DISCUSSION THREAD

**Discussion by shrimp9469** [jxlsteam/jxls#114] [Q&A] [answered]
Hello, 
       I used the .xlsx template to export about 55 columns of nearly 60,000 rows of data. I found that the memory usage reached about 7G, and it was not automatically released for a period of time after normal export. 
      Excuse me, is the memory footprint normal? Is there a way to solve the memory footprint and release problems?

**Comment by SoltauFintel** [comment, marked as answer]
POI uses XML objects for the in-memory representation of an XLSX file. That takes up a lot of memory.

It is recommended to use JXLS streaming.

For #59 I have to investigate where the limits are for the number of columns.

**Reply by shrimp9469** [reply]
Thank you very much. I'll try.

**Comment by nguacon90** [comment]
I think that you can write multiple excel files with limited row number for example 20.000 rows, also you get about 3 files excel, after that you merge 3 files into one.
